cmd/gossamer/config.go

```diff
@@ -127,10 +138,33 @@ func setDotAccountConfig(ctx *cli.Context, cfg *dot.AccountConfig) {
 
 // setDotCoreConfig sets dot.CoreConfig using flag values from the cli context
 func setDotCoreConfig(ctx *cli.Context, cfg *dot.CoreConfig) {
+	// check --roles flag and update node configuration
+	if roles := ctx.GlobalString(RolesFlag.Name); roles != "" {
+		// convert string to byte
+		b, err := strconv.Atoi(roles)
+		if err != nil {
+			log.Error("[cmd] Failed to convert Roles to byte", "error", err)
+		} else if byte(b) == 4 {
+			// if roles byte is 4, act as an authority (see Table D.2)
+			log.Debug("[cmd] Authority enabled", "roles", 4)
+			cfg.Authority = true
+		} else {
+			// if roles byte is not 4, do not act as an authority (see Table D.2)
+			log.Debug("[cmd] Authority disabled", "roles", byte(b))
+			cfg.Authority = false
+		}
+	}
+
 	// check --authority flag and update node configuration
 	if authority := ctx.GlobalBool(AuthorityFlag.Name); authority {
```

Comment:
we can probably just get rid of the authority flag

Reply:
was not sure if we were attached to it

Reply:
nah, feel free to remove